server/s3_factory.h

```diff
@@ -41,23 +41,23 @@ class S3BucketMetadataFactory {
  public:
   virtual ~S3BucketMetadataFactory() {}
   virtual std::shared_ptr<S3BucketMetadata> create_bucket_metadata_obj(
-      std::shared_ptr<S3RequestObject> req) {
+      std::shared_ptr<S3RequestObject> req, std::string bucket = "") {
```

Comment:
Did you update appropriate UTs?
MOCK_METHODn ->MOCK_METHOD(n+1)

Reply:
Yes